### PR TITLE
juiced changes

### PR DESCRIPTION
**Pull request by juiceddurian**

### CODE REVIEW

File: ts-client/package.json [hunk 1]
```diff
@@ -1,9 +1,11 @@
 {
-  "name": "sdk-ts",
+  "name": "@juiced-fi/mercurial-sdk",
```
Comment (dongnguyen9186) [inline ts-client/package.json:2]: can you please change it to `mercurial-sdk` ?

File: ts-client/.npmrc [hunk 2]
```diff
@@ -0,0 +1 @@
+@juiced-fi:registry=https://npm.pkg.github.com
```
Comment (dicksonpys) [inline ts-client/.npmrc:1]: what's the usage of this?

File: ts-client/src/vault.ts [hunk 3]
```diff
@@ -36,22 +36,22 @@ class Vault {
   }
 
   calculateLockedProfit(currentTime) {
-    const duration = currentTime - this.state.lockedProfitTracker.lastReport;
+    const duration = currentTime - this.state!.lockedProfitTracker.lastReport;
```
Comment (dicksonpys) [inline ts-client/src/vault.ts:39]: in favor of negate-first

```
calculateLockedProfit(currentTime) {
  if (!this.state) return null;
}
```

File: ts-client/tsconfig.json [hunk 4]
```diff
@@ -1,11 +1,25 @@
...
+    "module": "commonjs",
+    "moduleResolution": "node",
+    "resolveJsonModule": true,
+    "rootDir": "src",
+    "outDir": "./lib",
+    "declaration": true
+  },
+  "include": ["**/*.ts"],
+  "exclude": ["node_modules", "lib", "mango-client-v3", "tests"],
+  "types": "./types/*.d.ts"
```
Comment (dicksonpys) [inline ts-client/tsconfig.json:24]: 1. do we need to target es2020? I don't mind, just wondering if there will be compatibility issue
2. allowJs, we don't need this?

**Comment by andrewsource147** [comment]
I will close PR as our discussion

**Pull request closed without merging**